### PR TITLE
[incubator-kie-issues-1131] test migration from V7 to code generation-28

### PR DESCRIPTION
**Pull request by Abhitocode**
Below are the tests with issues, commented the same for each test.

**1. CompensationTest**

- **compensationInSubSubProcesses**
It appears there are a few nodes that are missing outbound or inbound connections. The BPMN editor didn't indicate any problems, but it turns out that these nodes have no connection to the start node.

error:
java.lang.IllegalStateException: Process could not be validated ![Process 'Default Process' [InSubSubProcess]: Node 'Boundary Compensate 3' [_X-1] Event has no outgoing connection
        Process 'Default Process' [InSubSubProcess]: Node 'Compensate Handler 3' [_X-2] Action has no incoming connection.,
        Process 'Default Process' [InSubSubProcess]: Node 'Boundary Compensate 2' [_Y-1] Event has no outgoing connection,
        Process 'Default Process' [InSubSubProcess]: Node 'Compensate Handler 2' [_Y-2] Action has no incoming connection.,
        Process 'Default Process' [InSubSubProcess]: Node 'Compensate Handler 3' [_X-2] Has no connection to the start node.,
        Process 'Default Process' [InSubSubProcess]: Node 'Compensate Handler 2' [_Y-2] Has no connection to the start node.]*

**2. DataTest**
- **testDataStore and testAssociation**
When compiling the code, we cannot resolve method 'metaData' in 'Process'.
As I understand, the metaData() method is protected, and thus, it isn't directly accessible in the test class.
How to get process metadata from the process object(only then we can get the definition metadata).

- **testDefaultProcessVariableValue** 
CorrelationKeyModel does not have methods named getProcVar and getIntVar .
This may be due to acutal variables/methods being different in the CorrelationKeyModel.

guidance on above would be helpful.

Closes https://github.com/apache/incubator-kie-issues/issues/1131.

Note: This pr contains tests with error, only for review purpose.

### CODE REVIEW

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java [hunk 1]
```diff
@@ -1903,90 +1911,84 @@ public void testBusinessRuleTaskWithExpressionsForIO() throws Exception {
         assertThat(account.getPerson()).isNotNull();
     }
 
+
     @Test
     @Disabled("this was not working")
     public void testUserTaskWithAssignment() throws Exception {
+        Application app = ProcessTestHelper.newApplication();
```
Comment (elguardian) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java:1918]: this cannot be migrated. remove the change

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java [hunk 2]
```diff
@@ -1456,20 +1462,22 @@ public void testServiceTaskWithMvelJaxbTransformation() throws Exception {
...
-
-        variables.put("tipoEvento", "error");
-        variables.put("pasoVariable", 3);
-        KogitoProcessInstance processInstance = kruntime.startProcess("ErrorsBetweenProcess", variables);
-
-        assertProcessInstanceCompleted(processInstance.getStringId(), kruntime);
-        assertProcessInstanceAborted(processInstance.getStringId() + 1, kruntime);
-
-        assertProcessVarValue(processInstance, "event", "error desde Subproceso");
+        Application app = ProcessTestHelper.newApplication();
```
Comment (elguardian) [inline jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java:1465]: you need to declare both process 

```
        Application app = ProcessTestHelper.newApplication();
        ProcessTestHelper.registerProcessEventListener(app, new DefaultKogitoProcessEventListener() {
            @Override
            public void afterProcessCompleted(ProcessCompletedEvent event) {
                if ("ErrorsBetweenSubProcess".equals(event.getProcessInstance().getProcessId())) {
                    assertThat(event.getProcessInstance().getState()).isEqualTo(ProcessInstance.STATE_ABORTED);
                }
            }
        });
        ErrorsBetweenSubProcessProcess.newProcess(app);
        org.kie.kogito.process.Process<ErrorsBetweenProcessModel> process = ErrorsBetweenProcessProcess.newProcess(app);
        ErrorsBetweenProcessModel model = process.createModel();
        model.setTipoEvento("error");
        model.setPasoVariable(3);
        ProcessInstance<ErrorsBetweenProcessModel> processInstance = process.createInstance(model);
        processInstance.start();
        assertThat(processInstance.status()).isEqualTo(ProcessInstance.STATE_COMPLETED);

        Assertions.assertEquals("error desde Subproceso", processInstance.variables().getEvent());
```

**Review by elguardian — CHANGES_REQUESTED** [review]
please fix

File: jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java (hunk 2 above)
Comment (Abhitocode) [reply jbpm/jbpm-tests/src/test/java/org/jbpm/bpmn2/ActivityTest.java:1465]: Migrated as above and created a separate pr for the same https://github.com/apache/incubator-kie-kogito-runtimes/pull/3665. Thank you @elguardian : )

**Pull request closed without merging**